aff3ct fails to install as a submodule

Hi,

I try to build my own Python bindings for aff3ct with pybind11. The examples in [aff3ct example project](https://github.com/aff3ct/my_project_with_aff3ct) suggest to link statically against libaff3ct. Also, they suggest to not install aff3ct but just copy the generated cmake.

In order to make the installation of my bindings easier, I'd like to include aff3ct as a submodule. My workflow would look smth like this:
```
git checkout --recursive URL
mkdir build
cd build
cmake ..
make -j[some_value]
make install
```
The last step in this procedure causes issues.

```
CMake Error at lib/aff3ct/cmake_install.cmake:64 (file):
  file INSTALL cannot find
  "{source_prefix}/pybind11_aff3ct/build/lib/cmake/aff3ct-2.3.5/aff3ct-config-version.cmake":
  No such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:63 (include)
```
I assume this is the case because of these lines in CMake:
https://github.com/aff3ct/aff3ct/blob/1ceddfc874f38317e83592b506b1fe78dffe44b4/CMakeLists.txt#L556-L558

I left the aff3ct library in the subfolder `lib/aff3ct` like in the example project. aff3ct CMake expects a file in my build folder at:
`lib/cmake/aff3ct-2.3.5/aff3ct-config-version.cmake` but it ends up in `lib/aff3ct/lib/cmake/aff3ct-2.3.5/aff3ct-config-version.cmake`.

It would be sufficient to change `CMAKE_BINARY_DIR` to `CMAKE_CURRENT_BINARY_DIR` to fix this issue.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -587,7 +587,7 @@ if (AFF3CT_COMPILE_SHARED_LIB OR AFF3CT_COMPILE_STATIC_LIB)
             NAMESPACE aff3ct::
             COMPONENT library)
 
-    install(FILES "${CMAKE_BINARY_DIR}/lib/cmake/aff3ct-${AFF3CT_VERSION_FULL}/aff3ct-config-version.cmake"
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/aff3ct-${AFF3CT_VERSION_FULL}/aff3ct-config-version.cmake"
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aff3ct-${AFF3CT_VERSION_FULL}/"
             COMPONENT library)
 

